### DISCUSSION TITLE
Solax X3 Pro modbus RTU RS485 support  (

### DISCUSSION THREAD

**Discussion by Meizel** [wills106/homeassistant-solax-modbus#549] [Ideas]
Is it possible to get support for the solax-x3-pro-xxK-g2?
Serial: MPT
Inverter: X3-PRO-25K-G2

Python script on github
[https://github.com/wesleyrijks/SolaxX3RS485-homeassistant/tree/main](url)

I'll like to have support for the pro version.
On tweakers.net [https://gathering.tweakers.net/forum/list_messages/2104298/0](url) dev5000 got this to work.
"forum is in Dutch" and after some googling i fond the github page wit the python script.

the mapping is completely different than  all the other inverters from solax. 
in the __init__.py you can find the address list for all the status for the sensors.

Greeting Dennis

**Comment by wills106** [comment]
This Integration already supports the MIC/MIC PRO G1/G2 I might just need to add the starting of your Serial Number in:

https://github.com/wills106/homeassistant-solax-modbus/blob/ebf5a81fb93f0bef5b2a8f61679e77ea576d4be6/custom_components/solax_modbus/plugin_solax.py#L5326-L5345
